### PR TITLE
fix(client): Added relevant placeholder to link submission

### LINKED ISSUE #38368: Unclear instructions with the use of Glitch to submit challenges.
I was recently attempting to submit my code changes of the [APIs and Mcroservices certification ](https://www.freecodecamp.org/learn/apis-and-microservices/managing-packages-with-npm/) through the URL of glitch as instructed in the introduction but I was finding trouble with passing the tests because I was submitting a wrong URL. Instead of something like  `https://blush-tundra-dodo.glitch.me/` I was submitting  `https://glitch.com/edit/#!/blush-tundra-dodo?path=package.json:7:21` until I was guided on the help forum before I got demotivated.

Don't you think we should revise the introduction and include steps on how to submit in the URLs from glitch to save future cases like mine? How do you see?

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38368

Changed logic on backend `Show.js` to incorporate useful placeholders based on `challengeType`.

### CODE REVIEW

**Review by hassaanp — APPROVED** [review]

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Individual challenges don't store the `githubLink`, that's only for projects.  If we use `ProjectForm` as is, then it will allow submission of Github links, which are not used.  As such, I see two good options.

1) just change the placeholder
2) add an 'isProject' prop to `ProjectForm` and have it handle all three cases (frontend project, backend project and backend challenge).  Maybe rename it to `SolutionForm` in that case.

Either would be fine with me, though obviously 1) is lot less work!

**Comment by Sky020** [comment]
> Individual challenges don't store the `githubLink`, that's only for projects. If we use `ProjectForm` as is, then it will allow submission of Github links, which are not used. As such, I see two good options.
> 
> 1. just change the placeholder
> 2. add an 'isProject' prop to `ProjectForm` and have it handle all three cases (frontend project, backend project and backend challenge).  Maybe rename it to `SolutionForm` in that case.
> 
> Either would be fine with me, though obviously 1) is lot less work!

Hey @ojeytonwilliams , I do not understand the issue. I do not see `githubLink` being passed between components. I can see the issue of the Glitch URL being the only required URL, but that is it.

**Comment by ojeytonwilliams** [comment]
> Hey @ojeytonwilliams , I do not understand the issue. I do not see `githubLink` being passed between components. I can see the issue of the Glitch URL being the only required URL, but that is it.

Hi @Sky020.  The problem is that if you go to a normal challenge, like [the one in the original issue](https://www.freecodecamp.org/learn/apis-and-microservices/managing-packages-with-npm/add-a-description-to-your-package.json) you should see a single input element to submit the solution. with a Glitch placeholder.  Reason being, Github links are used exclusively for backend projects (not backend challenges).  See https://github.com/freeCodeCamp/freeCodeCamp/blob/f25d035e9dcc38cde33a09436afb11dcbad7992b/client/src/templates/Challenges/redux/completion-epic.js#L86-L88

Since we only want the Glitch URL, we shouldn't ask users to submit anything else.

**Comment by Sky020** [comment]
Thank you, @ojeytonwilliams , for all your input. I believe all changes have been made.

**Comment by ojeytonwilliams** [comment]
@Sky020  no problem.

Unfortunately, this isn't quite what was after.  The challenges should only have the 'Solution' input element and not 'Github', since only the solution actually gets used.  If you look at [this challenge](https://www.freecodecamp.org/learn/apis-and-microservices/managing-packages-with-npm/how-to-use-package.json-the-core-of-any-node.js-project-or-npm-package), this is correct except for the placeholder.  Adding a Github input element, as the PR currently does, will lead people to think they need to submit their Github repository, which is not the case.

In other words what I think should happen is that placeholder should become `Link, ex: https://camperbot.glitch.me` and nothing else should change.

**Comment by Sky020** [comment]
> @Sky020 no problem.
> 
> Unfortunately, this isn't quite what was after. The challenges should only have the 'Solution' input element and not 'Github', since only the solution actually gets used. If you look at [this challenge](https://www.freecodecamp.org/learn/apis-and-microservices/managing-packages-with-npm/how-to-use-package.json-the-core-of-any-node.js-project-or-npm-package), this is correct except for the placeholder. Adding a Github input element, as the PR currently does, will lead people to think they need to submit their Github repository, which is not the case.
> 
> In other words what I think should happen is that placeholder should become `Link, ex: https://camperbot.glitch.me` and nothing else should change.

The most recent commit does make only the _Solution_ input available for challenges. Did I miss something?

**Comment by ojeytonwilliams** [comment]
When I open this in Gitpod I see two input elements image

I think the easiest thing would be just to just edit 
https://github.com/freeCodeCamp/freeCodeCamp/blob/a7c70ddcd82a59b17991f91707ae964bbe649744/client/src/templates/Challenges/projects/backend/Show.js#L93

that should change the placeholder for all the backend challenges.

The refactoring I suggested isn't critical, it's just nice to have.

**Comment by RandellDawson** [comment]
@Sky020 You need to rebase against master.  You have way more files with changes than necessary.  The PR should only show the files that you have changed.

**Comment by Sky020** [comment]
@RandellDawson , Sorry about the mess. I guess it was too late at night for me to be making commits. I will fix it as soon as I can.

**Comment by Sky020** [comment]
Closing pull-request. Corrected issue with this pull request: #38411 
Thank you, all, for your time and patience.
